pkg/state/infra.go

```diff
@@ -86,28 +139,54 @@ func (i *infraClient) updateBrokers(hosts []string) {
 		brokerState := i.brokerStateFactory(host, 5671)
 		i.brokers[host] = brokerState
 	}
+	return nil
 }
 
-func (i *infraClient) SyncConnectors(routers []string, brokers []string) ([]ConnectorStatus, error) {
+func (i *infraClient) SyncAll(routers []string, brokers []string) ([]ConnectorStatus, error) {
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
+	ctx := context.Background()
+
 	i.updateRouters(routers)
-	i.updateBrokers(brokers)
+
+	err := i.updateBrokers(ctx, brokers)
+	if err != nil {
+		return nil, err
+	}
 
 	// Ensure all routers are initialized
-	for _, router := range i.routers {
-		err := router.Initialize()
-		if err != nil {
-			return nil, err
-		}
+	err = i.applyRouters(ctx, func(router *RouterState) error {
+		return router.Initialize()
+	})
+
+	if err != nil {
+		log.Println("Error initializing routers", err)
+		return nil, err
+	}
+
+	// Ensure all brokers are initialized
+	err = i.applyBrokers(ctx, func(broker *BrokerState) error {
+		return broker.Initialize()
+	})
+	if err != nil {
+		log.Println("Error initializing brokers", err)
+		return nil, err
 	}
 
-	connectorStatuses := make([]ConnectorStatus, 0)
+	// Sync all known addresses and endpoints
+	err = i.syncConfiguration()
+	if err != nil {
+		return nil, err
+	}
 
+	totalConnectors := len(i.routers) * len(i.brokers)
+	connectorStatuses := make(chan ConnectorStatus, totalConnectors)
```

Comment:
why is the use of the channel helpful here?  I think everything is on the same thread